subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -28,6 +28,12 @@ import static org.junit.Assert.assertTrue
 @Unroll
 class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
 
+    public static final String COPY_UNREADABLE_DESTINATION_DEPRECATION = "Cannot access a file in the destination directory (see --info log for details). " +
+        "Copying to a directory which contains unreadable content has been deprecated. " +
+        "This will fail with an error in Gradle 8.0. " +
+        "Use the method Copy.ignoreExistingContentInDestinationDir(). " +
+        "Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_7.html#deprecate_unreadable_output"
```

Comment:
Why declare this right here? The choice seems arbitrary. I wonder if we could declare some common ancestor type instead where it can be declared.

Reply:
I removed the constant to a fixture.